test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -0,0 +1,202 @@
+	perNode = flag.Int("per_node", 1, "Number of hostname pods per node")
+)
+
+const (
+	podStartTimeout = 5 * time.Minute
+)
+
+func main() {
+	flag.Parse()
+
+	settings, err := clientcmd.LoadFromFile(filepath.Join(os.Getenv("HOME"), ".kube", ".kubeconfig"))
+	if err != nil {
+		log.Fatalf("Error loading configuration: %v", err.Error())
+	}
+	config, err := clientcmd.NewDefaultClientConfig(*settings, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		log.Fatalf("Failed to construct config: %v", err)
+	}
+
+	c, err := client.New(config)
+	if err != nil {
+		log.Fatalf("Failed to make client: %v", err)
+	}
+
+	nodes, err := c.Nodes().List()
+	if err != nil {
+		log.Fatalf("Failed to list nodes: %v", err)
+	}
+
+	if len(nodes.Items) == 0 {
+		log.Fatalf("Failed to find any nodes.")
+	}
+
+	fmt.Printf("Nodes found on this cluster:\n")
+	for i, node := range nodes.Items {
+		fmt.Printf("%d: %s\n", i, node.Name)
+	}
+
+	// Make a unique namespace for this test.
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	ns := "serve-hostnames-" + strconv.Itoa(r.Int()%10000)
+	fmt.Printf("Using namespace %s for this test.\n", ns)
+
+	// Put serve-hostname pods on each node.
+	podNames := []string{}
+	for i, node := range nodes.Items {
+		for j := 0; j < *perNode; j++ {
+			podName := fmt.Sprintf("serve-hostname-%d-%d", i, j)
+			podNames = append(podNames, podName)
+			fmt.Printf("Creating pod %s/%s on node %s ...", ns, podName, node.Name)
+			_, err := c.Pods(ns).Create(&api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name: podName,
+					Labels: map[string]string{
+						"name": "serve-hostname",
+					},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "serve-hostname",
+							Image: "kubernetes/serve_hostname:1.1",
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+					Host: node.Name,
+				},
+			})
+			if err != nil {
+				log.Fatalf("Failed to create pod server-hostname-%d", i)
+			}
+			fmt.Print(" [done]\n")
+		}
+	}
+	// Clean up the pods
+	defer func() {
+		for _, podName := range podNames {
+			if err = c.Pods(ns).Delete(podName); err != nil {
+				log.Fatalf("Failed to delete pod %s: %v", podName, err)
+			}
+		}
+	}()
+
+	// Create a service for these pods.
+	fmt.Print("Creating service serve-hostnames ...")
+	svc, err := c.Services(ns).Create(&api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name: "serve-hostnames",
+			Labels: map[string]string{
+				"name": "serve-hostname",
+			},
+		},
+		Spec: api.ServiceSpec{
+			Port:       9376,
+			TargetPort: util.NewIntOrStringFromInt(9376),
+			Selector: map[string]string{
+				"name": "serve-hostname",
+			},
+		},
+	})
+	if err != nil {
+		log.Fatalf("unable to create service %s/%s: %v", ns, svc.Name, err)
+	}
+	fmt.Print(" [done]\n")
+	// Clean up service
+	defer func() {
+		if err = c.Services(ns).Delete(svc.Name); err != nil {
```

Comment:
Thanks, done. I did think about these two `err` variables and somehow thought it was good to reuse to existing in scope `err` but I am wrong.